common/persistence/execution_manager.go

```diff
@@ -121,6 +120,30 @@ func (m *executionManagerImpl) UpdateWorkflowExecution(
 	request *UpdateWorkflowExecutionRequest,
 ) (*UpdateWorkflowExecutionResponse, error) {
 
+	var updateWorkflowNewEvents []*InternalAppendHistoryNodesRequest
+	var newWorkflowNewEvents []*InternalAppendHistoryNodesRequest
+	for _, workflowEvents := range request.CurrentWorkflowEvents {
+		newEvents, err := m.serializeWorkflowEvents(request.ShardID, workflowEvents)
+		if err != nil {
+			return nil, err
+		}
+		if newEvents != nil {
```

Comment:
why checking newEvents not being nil?
i believe `newEvents == nil` will be a bug?